geiger/src/geiger_syn_visitor.rs

```diff
@@ -0,0 +1,130 @@
+use super::{
+    file_forbids_unsafe, is_test_fn, is_test_mod, IncludeTests, RsFileMetrics,
+};
+
+use syn::{visit, Expr, ImplItemMethod, ItemFn, ItemImpl, ItemMod, ItemTrait};
+
+pub struct GeigerSynVisitor {
+    /// Count unsafe usage inside tests
+    pub include_tests: IncludeTests,
+
+    /// The resulting data from a single file scan.
+    pub metrics: RsFileMetrics,
+
+    /// The number of nested unsafe scopes that the GeigerSynVisitor are
+    /// currently in. For example, if the visitor is inside an unsafe function
+    /// and inside an unnecessary unsafe block inside that function, then this
+    /// number should be 2. If the visitor is outside unsafe scopes, in a safe
+    /// scope, this number should be 0.
+    /// This is needed since unsafe scopes can be nested and we need to know
+    /// when we leave the outmost unsafe scope and get back into a safe scope.
+    pub unsafe_scopes: u32,
+}
```

Comment:
Can we keep the members on this struct private or will that result in too much bloat?

Reply:
Ahh, it looks like the `include_tests` and `unsafe_scopes` can definitely be (my bad being too keen with the refactoring). I think the `metrics` needs to be public at the crate level (it's called in the newly created `find` module), but it should be restricted to that by https://github.com/rust-secure-code/cargo-geiger/pull/158/files#diff-a49da58edbe4631a734dcd7884f75469e5ee8ab7859659643656a113c9b2ee81R11 however I'm more than happy to add a `pub crate` restriction to it, to make explicit the exposure level?

Reply:
> I'm more than happy to add a pub crate restriction to it, to make explicit the exposure level?

Not needed on my opinion. The reason I wanted to restrict the other members was to avoid accidental coupling between different modules down the line. Keeping internal API:s inside a library clear and minimal tends to make maintenance and refactoring easier in the long run, in my experience.